Model/articles.model.js

```diff
@@ -13,17 +13,36 @@ exports.selectArticle = (articleID) => {
     })
 }
 
-exports.selectAllArticles = () => {
-    return db.query(
-        `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles
+exports.selectAllArticles = (topic) => {
+   
+    let sqlQuery = 
+    `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles
         LEFT JOIN comments
-        ON comments.article_id = articles.article_id
+        ON comments.article_id = articles.article_id `
+
+    if(topic){
+        return db.query(
+        sqlQuery += 
+        `WHERE topic = $1
         GROUP BY articles.article_id
```

Comment:
You don't need to have the GROUP BY and ORDER BY repeated twice here - if there is a topic you could just ad the WHERE clause. Whether there is a topic or not, we are always going to have the GROUP BY AND ORDER BY.